src/main/java/com/nha/abdm/wrapper/hip/hrp/database/mongo/services/PatientService.java

```diff
@@ -220,4 +221,35 @@ public FacadeResponse upsertPatients(List<Patient> patients) {
             String.format("Successfully upserted %d patients", bulkWriteResult.getUpserts().size()))
         .build();
   }
+
+  /**
+   * <B>Data Transfer</B>
+   * For a given list of careContextsReference and patientReference check whether careContexts match with patient.
+   * @param careContextsWithPatientReference Has CareContextReference and patientReference.
+   * @return if all the Contexts match with respective patient return true;
+   */
+  public boolean isCareContextPresent(List<CareContextsWithPatientReference> careContextsWithPatientReference) {
+    if (careContextsWithPatientReference == null) return false;
+    for (CareContextsWithPatientReference careContexts : careContextsWithPatientReference) {
+      Patient patient = patientRepo.findByPatientReference(careContexts.getPatientReference());
+      if (patient == null) {
+        return false;
+      }
+      List<CareContext> existingCareContexts = patient.getCareContexts();
+      if (existingCareContexts == null) return false;
+      if (!existingCareContexts.stream()
+              .anyMatch(
```

Comment:
This looks like even if one of the `careContext` is present then it returns true. What if the careContexts match partially?

Reply:
As of now, partial careContexts are not implemented because, for every list of careContexts there is a single consentId, which is used as a communication with ABDM, so I'm honoring the consentId instead of careContexts. open for discussion